src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -31,6 +31,33 @@ public interface ExternalObjectDirectoryRepository extends JpaRepository<Externa
     List<ExternalObjectDirectoryEntity> findByMediaStatusAndType(MediaEntity media, ObjectRecordStatusEntity status,
                                                                  ExternalLocationTypeEntity externalLocationType);
 
+    @Query(
+        """
+                SELECT eod FROM ExternalObjectDirectoryEntity eod
+                WHERE eod.media = :media
+                AND eod.status = :status
+            """
+    )
+    List<ExternalObjectDirectoryEntity> findByEntityAndStatus(MediaEntity media, ObjectRecordStatusEntity status);
+
+    @Query(
+        """
+                SELECT eod FROM ExternalObjectDirectoryEntity eod
+                WHERE eod.annotationDocumentEntity = :annotationDocumentEntity
+                AND eod.status = :status
+            """
+    )
+    List<ExternalObjectDirectoryEntity> findByEntityAndStatus(AnnotationDocumentEntity annotationDocumentEntity, ObjectRecordStatusEntity status);
```

Comment:
i deliberately done this so that the methods are overloaded. I can change it if you feel its worth it

Reply:
I think so, line 50 and 59 stand out to me when reading the code - that's why I commented.
It would be simpler to adopt one style, maybe like on line 41.
But appreciate it means the query params would need to be changed also.
I'll leave it with you to decide to update or not sure.  Thanks

Reply:
To be clear, my comment was around the method param names, not the name of the method itself

Reply:
i had misunderstood, think i've it changed now.